nixos/modules/services/networking/headscale.nix

```diff
@@ -0,0 +1,473 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.headscale;
+
+  dataDir = "/var/lib/headscale";
+  runDir = "/run/headscale";
+
+  settingsFormat = pkgs.formats.yaml { };
+  configFile = settingsFormat.generate "headscale.yaml" cfg.extraSettings;
+in
+{
+  options = {
+    services.headscale = {
+      enable = mkEnableOption "headscale, Open Source coordination server for Tailscale";
+
+      package = mkPackageOption pkgs "headscale";
+
+      user = mkOption {
+        default = "headscale";
+        type = types.str;
+        description = ''
+          User the headscale server should execute under.
+        '';
```

Comment:
```suggestion
        description = ''
          User account under which headscale runs.
          <note><para>
          If left as the default value this user will automatically be created
          on system activation, otherwise you are responsible for
          ensuring the user exists before the headscale service starts.
          </para></note>
        '';
```

Reply:
there was a discussion a while back about maybe adding a `mkUserOption`/`mkGroupOption` with such a text. feels like an increasingly good idea to standardize this :)

Reply:
mm, that sounds sensible!

Reply:
For sure! But in the meantime let's not make good the enemy of perfect.

Reply:
for sure! just as a reminder in case anyone wants to pick that up :D